plugins/overwatch.rb

```diff
@@ -0,0 +1,660 @@
+    # @netmask [string array] route netmask
+    # @curr_sid [int class] Session ID of the current session being processed
+    # @stale_sids [int array] Array of session ID's that have been marked as stale.
+    #
+    # @return [void] A useful return value is not expected here
+    def remove_if_stale(subnet, netmask, curr_sid, stale_sids)
+      Rex::Socket::SwitchBoard.each do | route |
+        if stale_sids.include? route.comm.sid # See if a route is associated with a stale session
+          if route.subnet == subnet && route.netmask == netmask # See if that route matches the one currently being processed
+            session_log(route.comm.sid, "Deleting route #{route.subnet}/#{route.netmask}")
+            begin
+              Rex::Socket::SwitchBoard.remove_route(route.subnet, route.netmask, route.comm) # Remove so fresh matching route can be added
+            rescue ::Exception => error
+              session_log(route.comm.sid, "Could not remove route to subnet #{route.subnet}/#{route.netmask}")
+              session_log(route.comm.sid, error.to_s)
+            end
+            return
+          end
+        end
+      end
+    end
+
+    # Take an IP address and a netmask and return the appropreate subnet "Network"
+    #
+    # @ip_addr [string class] Input IPv4 Address
+    # @netmask [string class] Input IPv4 Netmask
+    #
+    # @return [string class] The subnet related to the IP address and netmask
+    # @return [nil class] Something is out of range
+    def get_subnet(ip_addr, netmask)
+      return nil if !validate_cmd(ip_addr, netmask) #make sure IP and netmask are valid
+
+      nets = ip_addr.split('.')
+      masks = netmask.split('.')
+      output = ""
+
+      (0..3).each do | index |
+        octet = get_subnet_octet(int_or_nil(nets[index]), int_or_nil(masks[index]))
+        return nil if !octet
+        output << octet.to_s
+        output << '.' if index < 3
+      end
+      return output
+    end
+
+    # Input an octet of an IPv4 address and the cooresponding octet of the
+    # IPv4 netmask then return the appropreate subnet octet.
+    #
+    # @net  [integer class] IPv4 address octet
+    # @mask [integer class] Ipv4 netmask octet
+    #
+    # @return [integer class] Octet of the subnet
+    # @return [nil class] If an input is nil
+    def get_subnet_octet(net, mask)
+      return nil if !net || !mask
+      subnet_range = 256 - mask  # This is the address space of the subnet octet
+      multi = net / subnet_range # Integer division to get the multiplier needed to determine subnet octet
+      return(subnet_range * multi) # Multiply to get subnet octet
+    end
+
+    # Take a string of numbers and converts it to an integer.
+    #
+    # @string [string class] Input string, needs to be all numbers (0..9)
+    #
+    # @return [integer class] Integer representation of the number string
+    # @return [nil class] string contains non-numbers, cannot convert
+    def int_or_nil(string)
```

Comment:
Simply to make @egypt's skin crawl, i propose: _string.to_i.to_s == string ? string.to_i : nil_